### PR TITLE
Allow skipping results verification with and without pushdown

### PR DESCRIPTION
**Pull request by hashhar**
Supersedes https://github.com/trinodb/trino/pull/8025.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java [hunk 1]
```diff
@@ -308,6 +309,12 @@ public QueryAssert skippingTypesCheck()
             return this;
         }
 
+        public QueryAssert skipVerifyingResultsWithPushdownDisabled()
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java:312]: nit: `skipResultsCorrectnessCheckForPushdown` ?

File: core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java [hunk 2]
```diff
@@ -451,15 +456,19 @@ public final QueryAssert isNotFullyPushedDown(PlanMatchPattern retainedSubplan)
...
 
+            if (!skipVerifyingResultsWithPushdownDisabled) {
+                // Compare the results with pushdown disabled, so that explicit matches() call is not needed
+                verifyResultsWithPushdownDisabled();
+            }
             return this;
         }
 
-        private void verifyResultsWithPushdownDisabled()
+        private QueryAssert verifyResultsWithPushdownDisabled()
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java:466]: Doesn't need to be fluent.

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java (hunk 2 above)
Comment (hashhar) [reply core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java:466]: Oh yes, it's private. Leftover from initial attempt.

**Review by wendigo — APPROVED** [review]

**Review by wendigo — APPROVED** [review]
lgtm

**Review by losipiuk — APPROVED** [review]

**Comment by findepi** [comment]
```
Error:  init(io.trino.plugin.sqlserver.TestSqlServerWithSnapshotIsolation)  Time elapsed: 15.933 s  <<< FAILURE!
java.lang.RuntimeException: Failed to execute statement: ALTER DATABASE database_da73c2b1cb834b878c2f230c864d09b7 SET READ_COMMITTED_SNAPSHOT ON
```

is the only failure

**Pull request merged**